### PR TITLE
Refactor author templates

### PR DESCRIPTION
**Pull request by nick-mon1**
## Summary
Removed duplicate logic for authors template and combined into one partial.

### Solution
- Combined `get_authors_short.html` and `get_authors.html` into `authors-list.html` to display the author component. Replaced markup on `authors/terms.html` to use new component
- Replaced grid styles/markup with `media-block`
- Updated styles to use BEM naming and functions


> [!NOTE]
> No visual design changes are affected.

<details><summary>Screenshots</summary>
<p>

<img width="518" alt="Screenshot 2024-06-04 at 3 12 34 PM" src="https://github.com/GSA/digitalgov.gov/assets/104778659/df56b8c4-8a83-48dc-bdad-1b0c5f028754">

</p>
</details> 


### How To Test

1. Visit page below
2. No visual regression changes are displayed


### What To Test


| template             | preview                                                                                                                                                                                                                   | live                                                                                                |
| -------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------- |
| card-article.html    | [preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-author-templates/news/)                                                                    | [live](https://digital.gov/news/)                                                                   |
| card-event.html      | [preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-author-templates/events/)                                                                  | [live](https://digital.gov/events/)                                                                 |
| card-service.html    | [preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-author-templates/services/directory/)                                                      | [live](https://digital.gov/services/directory/)                                                     |
| event/single.html    | [preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-author-templates/event/2022/05/03/the-basics-of-buying-accessible-products-and-services/)  | [live](https://digital.gov/event/2022/05/03/the-basics-of-buying-accessible-products-and-services/) |
| news/single.html     | [preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-author-templates/2024/02/09/timeless-top-10-best-practices-for-great-government-websites/) | [live](https://digital.gov/2024/04/16/determining-the-true-value-of-a-website-a-gsa-case-study/)    |
| news_featured.html   | [preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-author-templates/)                                                                         | [live](https://digital.gov/)                                                                        |
| service-contact.html | [preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-author-templates/services/directory/)                                                      | [live](https://digital.gov/services/directory/)                                                     |
| authors/terms.html   | [preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-author-templates/authors/)                                                                 | [live](https://digital.gov/authors/)                                                                |


---

### Dev Checklist

- [ ] PR has correct labels
- [ ] A11y testing (voice over testing, meets WCAG, run axe tools)
- [ ] Code is formatted properly

### CODE REVIEW

File: themes/digital.gov/layouts/partials/core/get_authors.html [hunk 1]
```diff
@@ -1,63 +1,75 @@
...
+                  aria-label="Read more articles by {{ $displayName | default $uid }}"
+                >
+                  {{- $displayName | default $uid -}}
+                </a>
+                {{- if $agency -}}
+                  <p>
+                    {{- $agency -}}
+                  </p>
+                {{- end -}}
+              </div>
```
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/get_authors.html:69]: There isn't much difference from Short and Long variants. Are there other use cases?

image

File: themes/digital.gov/src/scss/new/_authors-list.scss
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/_authors-list.scss:1]: Can we avoid mixins for single properties?

**Review by mejiaj — CHANGES_REQUESTED** [review]
I like that we're refactoring and simplifying our components. There's a lot of logic in the templates for not much payoff. Hope we can continue to simplify.

Also, does this need to be merged before #7639?

File: themes/digital.gov/layouts/partials/core/get_authors.html [hunk 2]
```diff
@@ -1,63 +1,75 @@
...
+        {{ if $isShort }}
+          <a
+            class="author"
+            href="{{- (printf "authors/%s/" $uid) | absURL -}}"
+            rel="author"
+          >
+            {{- partial "core/get_author_image.html" (dict "author" . ) -}}
+            <span class="usa-sr-only">Posts by</span>
+            <span>{{- $displayName | default $uid | markdownify -}}</span> </a
+          >{{ if lt (add $i 1) $authors_count }}<!--,-->{{ end }}
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/get_authors.html:39]: No, this is removed.

File: themes/digital.gov/layouts/partials/core/get_authors.html [hunk 3]
```diff
@@ -1,63 +1,75 @@
...
+{{ if $context.Params.authors }}
+  <div class="authors-list {{ if $isShort }}authors-list-short{{ end }}">
+    {{- $authors_count := $context.Params.authors | len -}}
 
-      {{/* If the byline was entered as a display name (i.e. - John Smith )
-        then this will turn that into a UID for use in a URL.
-      */}}
+    {{- range $i, $e := $context.Params.authors -}}
+      {{/* Convert into UID (i.e. john-smith) if author name is set as display name (i.e. John Smith) */}}
       {{- $uid := $e | urlize -}}
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/get_authors.html]: Improved the documentation for explaining why we use `$uid`. It is the same value as `slug` which we cannot access before we call `GetPage` on the next line. Instead we use `.Params.authors` to lookup the author page.

File: themes/digital.gov/layouts/partials/core/get_authors.html [hunk 4]
```diff
@@ -1,63 +1,75 @@
...
-                <p>
-                  {{- $author.Params.agency -}}{{- if $author.Params.location -}}
-                    &nbsp;|&nbsp;{{- $author.Params.location -}}
-                  {{- end -}}
-                </p>
-              {{- end -}}
+        {{ if $isShort }}
+          <a
+            class="author"
+            href="{{- (printf "authors/%s/" $uid) | absURL -}}"
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/get_authors.html:35]: See comment above.

File: themes/digital.gov/layouts/partials/core/get_authors.html (hunk 1 above)
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/get_authors.html:72]: See the screenshot in the PR description above. The only difference is that `authors-list` displays as a vertical stacked list along with some additional spacing and layout style changes. See https://digital.gov/event/2022/07/20/white-house-and-gsa-open-innovation-forum-building-equitable-partnerships/ for a better example.

File: themes/digital.gov/src/scss/new/_authors-list.scss
Comment (nick-mon1) [reply themes/digital.gov/src/scss/new/_authors-list.scss:1]: Updated styles to use functions and added BEM classes to improve readability.

**Review by RileySeaburg — DISMISSED** [review]

File: themes/digital.gov/layouts/partials/core/get_authors.html (hunk 1 above)
Comment (mejiaj) [reply themes/digital.gov/layouts/partials/core/get_authors.html:72]: Is there enough difference in the code to merit its own components? For example `authors-stacked`?

File: themes/digital.gov/src/scss/new/_service-contact.scss [hunk 5]
```diff
@@ -85,7 +85,7 @@
       }
     }
 
-    .authors-list-short {
+    .authors-list--inline {
       @include u-margin(0);
     }
   }
```
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/_service-contact.scss:91]: Do we need this page override?

**Review by mejiaj — CHANGES_REQUESTED** [review]
Looking a lot better. 

Added some suggestions for improving maintainability and code quality.

File: themes/digital.gov/layouts/partials/core/authors_list.html [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  @example
+  {{- partial "core/authors_list.html" (dict "data" $authorData "isStacked" true)
+-}}
+*/}}
+{{ $data := .data }}
+{{ $isStacked := default false .isStacked }}
+{{ $authorClasses := slice (print (cond $isStacked "authors-list" "authors-list--inline")) }}
+
+{{ if $data }}
+  <div class="{{ delimit $authorClasses " " }}">
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/authors_list.html:19]: Modified the logic to always apply the base class and modifier when set.

File: themes/digital.gov/layouts/services/service-contact.html [hunk 7]
```diff
@@ -104,7 +104,15 @@ <h3>
...
-            {{- partial "core/get_authors_short.html" . -}}
+            {{ if .Params.authors }}
+              {{ $authors := where .Site.Pages "Section" "authors" }}
+              {{ $authorData := slice }}
+              {{ range $authorName := .Params.authors }}
+                {{ $authorData = $authorData | append ( where $authors ".Params.slug" "eq" . ) }}
+              {{ end }}
+              {{ partial "core/authors_list.html" (dict "data" $authorData) }}
+            {{ end }}
+
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/layouts/services/service-contact.html:115]: Used the existing pattern of making the queries on the template pages, but to your point it only adds more code. Refactored this back to existing pattern to call the authors' data in the component itself.

File: themes/digital.gov/src/scss/new/_service-contact.scss (hunk 5 above)
Comment (nick-mon1) [reply themes/digital.gov/src/scss/new/_service-contact.scss:91]: Yes, with the new `authors-list--inline` variant class, we need this:

Screenshot 2024-07-02 at 2 11 21 PM

File: themes/digital.gov/src/scss/new/_authors-list.scss [hunk 8]
```diff
@@ -2,53 +2,64 @@
...
-  @include u-margin-bottom(7);
+  margin-bottom: units(7);
+  margin-top: 0;
 
   .author {
-    @include u-margin-bottom("105");
-    @include u-padding-top("105");
-    @include u-border-top("1px", "solid", "gray-10");
-    @include u-flex("no-wrap");
+    border-top: 1px color("gray-10") solid;
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/src/scss/new/_authors-list.scss:11]: Good call, updated.

**Review by mejiaj — CHANGES_REQUESTED** [review]
Authors inside of components look good. I checked `/authors` and am noticing regressions. Are you seeing this?

[Authors preview →](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-author-templates/authors/) 

image

**Comment by nick-mon1** [comment]
### Issue

Hey @mejiaj,

Currently, code for the authors templates uses two ways to display author data, one with page objects on landing pages and the other as a list of string names passed in from other partials. We could modify the `authors_list.html` to handle both instances without changing the current component API.

Benefits of updating the partial to accept both would be less code to write when on landing pages for restructuring the data to work with `authors_list.html` and we use one partial for displaying all author information on the site. The cons would be adding complexity to the template logic but might be necessary due to the way the author templates lack consistency.

Some pseudocode for what this would look like:

```
  {{ $isStringList := false }}
  {{ $isPageObjectList := false }}
  
  {{ /* Determine if we're dealing with a string list or page object list */}}
  {{ if eq (type .items) "string" }}
    {{ $isStringList = true }}
  {{ elif eq (type (index .items 0)) "hugo.Page" }}
    {{ $isPageObjectList = true }}
  {{ end }}
  
  {{ /* Render items depending on their type */}}
  {{ if $isStringList }}
    {{ range $item := .items }}
      {{ partial "authors_list.html" . }}
    {{ end }}
  {{ elseif $isPageObjectList }}
    {{ range $page := .items }}
      {{ partial "authors_list.html" . }}
    {{ end }}
  {{ end }}
```


I favor the above option to simplify the markup and reduce duplication.
Any thoughts or concerns on going this direction?

**Comment by RileySeaburg** [comment]
@nick-mon1 am I misunderstanding something? Why do we need one complicated template to solve two distinct problems?

**Comment by mejiaj** [comment]
> Currently, code for the authors templates uses two ways to display author data, one with page objects on landing pages and the other as a list of string names passed in from other partials.

1. To confirm, author templates refer to page templates or the components under `partials/core` (i.e. `author_image.html` & `authors_list.html`)?
 
> We could modify the `authors_list.html` to handle both instances without changing the current component API.

2. I see `authors_list.html` looks for authors via parent scope. This is what you mean by modifying to handle both instances? 

    _For example, by creating an `$authors` param that can be overridden, but by default is the parent scope of authors._

> Benefits of updating the partial to accept both would be less code to write when on landing pages for restructuring the data to work with `authors_list.html` and we use one partial for displaying all author information on the site. The cons would be adding complexity to the template logic but might be necessary due to the way the author templates lack consistency.
> 
> Some pseudocode for what this would look like:
> 
> ```
>   {{ $isStringList := false }}
>   {{ $isPageObjectList := false }}
>   
>   {{ /* Determine if we're dealing with a string list or page object list */}}
>   {{ if eq (type .items) "string" }}
>     {{ $isStringList = true }}
>   {{ elif eq (type (index .items 0)) "hugo.Page" }}
>     {{ $isPageObjectList = true }}
>   {{ end }}
>   
>   {{ /* Render items depending on their type */}}
>   {{ if $isStringList }}
>     {{ range $item := .items }}
>       {{ partial "authors_list.html" . }}
>     {{ end }}
>   {{ elseif $isPageObjectList }}
>     {{ range $page := .items }}
>       {{ partial "authors_list.html" . }}
>     {{ end }}
>   {{ end }}
> ```
> 
> I favor the above option to simplify the markup and reduce duplication. Any thoughts or concerns on going this direction?

3. Is there any way to normalize the data so that the component takes an generic data object instead of a page/component specific object?

    **Pseudocode example**
    Using your variable names to avoid confusion, but I'd name them something more specific.
    ```
   <!-- On page scope →
   { partial "core/authors_list.html" (dict "scope" . "isInline" true "authors" $PageObjectList ) }}

   <!-- Component scope →
   { partial "core/authors_list.html" (dict "scope" . "isInline" true "authors" $StringList ) }}
    ```

File: themes/digital.gov/layouts/partials/core/author_image.html
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/author_image.html:1]: I've created a card in trello for related authors work and general logo and icon logic that is re-used through out the site.

File: themes/digital.gov/layouts/authors/terms.html [hunk 9]
```diff
@@ -1,11 +1,20 @@
 {{ define "content" }}
-  <main role="main" id="main-content">
+  {{/* Stores an array of string author names */}}
+  {{ $authors := (where .Site.Pages "Section" "authors") }}
+  {{ $authors_data := slice }}
+  {{ range $i, $e := $authors }}
+    {{ if ne $e.Params.slug nil }}
+      {{ $authors_data = $authors_data | append $e.Params.slug }}
+    {{ end }}
+  {{ end }}
```
Comment (nick-mon1) [inline themes/digital.gov/layouts/authors/terms.html:9]: Use an array of strings of author names for landing page.

File: themes/digital.gov/src/scss/new/_authors-list.scss [hunk 10]
```diff
@@ -2,53 +2,82 @@
...
+    @include u-line-height("sans", 1);
+    @include u-margin-y("05");
+    @include u-text("bold", "black", "no-underline");
 
-    img {
-      @include u-border("1px", "solid", "indigo-30");
+    &:hover {
+      @include u-border-bottom("2px", "accent-warm", "solid");
     }
+  }
```
Comment (nick-mon1) [inline themes/digital.gov/src/scss/new/_authors-list.scss:20]: Added styles for overriding component styles when on [/authors](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-author-templates/authors/)

File: themes/digital.gov/layouts/news/card-article.html
Comment (mejiaj) [inline themes/digital.gov/layouts/news/card-article.html:1]: **question:** What's this empty file for?

File: themes/digital.gov/layouts/news/single.html
Comment (mejiaj) [inline themes/digital.gov/layouts/news/single.html:1]: > [!NOTE]  
> This changes the mobile design.

| Before | After |
| :----- | :---- |
|   image | image | 

Authors no longer lists under title.

[Source →](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-author-templates/2024/07/08/evaluating-your-agency-accessibility-just-got-a-whole-lot-easier-with-gsa-openacr-editor/)

**Review by mejiaj — COMMENTED** [review]
Thanks for the updates. Found no visual regressions and `/authors` is fixed.

Small question on an empty template. I can approve PR once resolved.

File: themes/digital.gov/layouts/news/card-article.html
Comment (nick-mon1) [reply themes/digital.gov/layouts/news/card-article.html:1]: Hmmm, this looks to be some merge conflict artifact. This template was removed in favor of `card-news.html`. I've removed this file.

File: themes/digital.gov/layouts/news/single.html
Comment (nick-mon1) [reply themes/digital.gov/layouts/news/single.html:1]: Good catch, this secondary instance was removed initially and is only display on mobile. I've [added it back](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-author-templates/2024/07/08/evaluating-your-agency-accessibility-just-got-a-whole-lot-easier-with-gsa-openacr-editor/).

File: themes/digital.gov/layouts/partials/core/authors_list.html [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+  {{- partial "core/authors_list.html" (dict "scope" . "isInline" true)
+-}}
+*/}}
+{{ $scope := .scope }}
+{{ $isInline := default false .isInline }}
+{{/* Returns the list of authors from the parent scope */}}
+{{ $authors := $scope.Param "authors" }}
+
+{{ if $authors }}
+  <div class="authors-list {{ if $isInline }}authors-list--inline{{ end }}">
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/authors_list.html]: I need to find a hugo solution that uses this pattern for 2 classes. Currently we use this when we have 3 or more classes.

**Review by mejiaj — APPROVED** [review]
Thanks for updating

**Review by RileySeaburg — APPROVED** [review]
Nice and clean, great job @nick-mon1

**Pull request merged**